src/main/java/eu/europeana/apikey/keycloak/KeycloakUserManager.java

```diff
@@ -0,0 +1,253 @@
+     * @param userId          the id of the client that is to be deleted
+     */
+    public boolean deleteUser(String userId, KeycloakSecurityContext adminSecurityContext) {
+        HttpDelete httpDelete = new HttpDelete(KeycloakUriBuilder.fromUri(String.format(USER_ENDPOINT,
+                                                                                        kcProperties.getAuthServerUrl(),
+                                                                                        kcProperties.getRealm(),
+                                                                                        userId)).build());
+        addAuthorizationHeader(adminSecurityContext.getAccessTokenString(), httpDelete);
+        try (CloseableHttpResponse response = httpClient.execute(httpDelete)) {
+            LOG.debug("Received response for user {} from Keycloak: {}", httpDelete, response);
+            if (response.getStatusLine().getStatusCode() != HttpStatus.NO_CONTENT.value()) {
+                return false;
+            }
+        } catch (IOException e) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Configure get request for retrieving the User identified by userId from Keycloak
+     *
+     * @param userId     api key used as keycloak userId
+     * @param adminToken access token to authorize request
+     * @return configured HttpGet request
+     */
+    private HttpGet prepareGetUserRequest(String userId, String adminToken) {
+        HttpGet httpGet = new HttpGet(KeycloakUriBuilder.fromUri(String.format(USER_ENDPOINT,
+                                                                               kcProperties.getAuthServerUrl(),
+                                                                               kcProperties.getRealm(),
+                                                                               userId)).build());
+        addAuthorizationHeader(adminToken, httpGet);
+        return httpGet;
+    }
+
+    /**
+     * Retrieve details about a given user identified by userId
+     * This method silently logs any errors that may occur communicating with Keycloak in order to allow
+     * a report message to be delivered to Slack
+     *
+     * @param httpGet get request
+     * @return a list of retrieved users
+     * @throws MissingKCUserException in case the user cannot be found
+     * @throws KCComException in case keycloak refuses to communicate
+     */
+    private UserRepresentation getUser(HttpGet httpGet, String userId) throws MissingKCUserException, KCComException {
+        LOG.debug("Sending user representation request to Keycloak...");
+        try (CloseableHttpResponse response = httpClient.execute(httpGet)) {
+            LOG.debug("Received user representation from Keycloak");
+            if (response.getStatusLine().getStatusCode() == HttpStatus.OK.value()) {
+                InputStream is = response.getEntity().getContent();
```

Comment:
Inputstream is never closed?

Reply:
This was marked as resolved, but I don't see how it is resolved? The CloseabeHttpResponse is closed, but I doubt that also closes the Inputstream